pkg/registry/populator.go

```diff
@@ -284,14 +286,17 @@ func (i *DirectoryPopulator) getNextReplacesImagesToAdd(imagesToAdd []*ImageInpu
 				pkgFoundImages++
 				foundImages = append(foundImages, pkgImage)
 			} else {
-				pkgRemainingImages++
-				remainingImages = append(remainingImages, pkgImage)
+				if !i.permissive {
+					pkgRemainingImages++
+					remainingImages = append(remainingImages, pkgImage)
```

Comment:
Are `remainingImages` not used by callers who want permissive population?

Reply:
@estroz `remainingImages` are the `pkgImage` s that are not "valid bundles" and could not be added to [`foundImages`](https://github.com/operator-framework/operator-registry/blob/0778a098b6827e60812eeecaf36d4eba25aa6587/pkg/registry/populator.go#L304). The reason we need to not add "invalid bundles" to `remainingImages` when permissive mode is enabled is to make sure that we don't [loop infinitely because `imagesToAdd`](https://github.com/operator-framework/operator-registry/blob/0778a098b6827e60812eeecaf36d4eba25aa6587/pkg/registry/populator.go#L185-L186) always has those "invalid images" in them